apps/mymedicare_cb/tests/test_models.py

```diff
@@ -390,9 +406,69 @@ def test_user_mbi_updated_from_different_value(self, mock_archive, mock_match_fh
         slsx_client.mbi = slsx_mbi
         slsx_client.hicn_hash = '50ad63a61f6bdf977f9796985d8d286a3d10476e5f7d71f16b70b1b4fbdad76b'
 
-        user, crosswalk_type = get_and_update_user(slsx_client, mock_request)
+        user, crosswalk_type = get_and_update_user_from_initial_auth(slsx_client, mock_request)
 
         user.refresh_from_db()
         crosswalk.refresh_from_db()
         self.assertEqual(user.crosswalk.user_mbi, slsx_mbi)
         mock_archive.assert_called_once()
+
+    @patch('apps.fhir.server.authentication.search_fhir_id_by_identifier', side_effect=search_fhir_id_by_identifier_side_effect)
+    @patch('apps.fhir.bluebutton.models.ArchivedCrosswalk.create')
+    def test_get_and_update_from_refresh_fhir_id_v3_previously_null(self, mock_archive, mock_match_fhir) -> None:
+        """Test that the get_and_update_from_refresh executes fields correctly,
+        specifically in this test, fhir_id_v3
+        """
+        user_hicn_hash = '50ad63a61f6bdf977f9796985d8d286a3d10476e5f7d71f16b70b1b4fbdad76b'
+        user_mbi = '1S00EU7JH00'
+        fhir_id_v2 = '-20140000008325'
+        username = '00112233-4455-6677-8899-aabbccddeeff'
+
+        fake_user = User.objects.create_user(
+            username=username,
+            email='fu@bar.bar'
+        )
+
+        crosswalk = Crosswalk.objects.create(
+            user=fake_user,
+            fhir_id_v2=fhir_id_v2,
+            user_hicn_hash=user_hicn_hash,
+            user_mbi=user_mbi,
+            user_id_type='M'
+        )
+        # Confirm fhir_id_v3 is None before calling the function
+        assert crosswalk.fhir_id_v3 is None
+
+        user, crosswalk_type = get_and_update_from_refresh(user_mbi, username, user_hicn_hash, mock_request)
```

Comment:
Do we have a test/want a test that asserts that the result of an update via each method produces the same result? E.g. something that calls one refresh, caches a result/value from the DB, then resets the DB (say, by `null`ing out the `v3` id), calls the other refresh, and then compares the two? Or, because of how we mock things, is that redundant/not meaningful?